xchange-gdax/src/main/java/org/knowm/xchange/gdax/service/GDAXMarketDataService.java

```diff
@@ -51,10 +56,67 @@ public OrderBook getOrderBook(CurrencyPair currencyPair, Object... args)
     return GDAXAdapters.adaptOrderBook(getGDAXProductOrderBook(currencyPair, level), currencyPair);
   }
 
+  /**
+   * Get trades data for a specific currency pair
+   * <p>
+   * If invoked with only the currency pair, the method will
+   * make a single api call, returning the default number 
+   * (currently 100) of the most recent trades.
+   * </p>
+   * If invoked with either optional argument the other must be
+   * specified as well.
+   *
+   * @param  currencyPair Currency pair to obtain trades for (required)
+   * @param  args[0] fromTradeId (Long) Return Trades with tradeIds greater than or equal to this value. Additional values may be returned. (optional)
+   * @param  args[1] toTradeId (Long) Return Trades with tradeIds up to but not including this value (optional)
+   * 
+   * @return A Trades object holding the requested trades
+   */
   @Override
   public Trades getTrades(CurrencyPair currencyPair, Object... args)
       throws IOException, RateLimitExceededException {
-
-    return GDAXAdapters.adaptTrades(getGDAXTrades(currencyPair), currencyPair);
+	  
+	  if (args.length == 0) {
+		  return GDAXAdapters.adaptTrades(getGDAXTrades(currencyPair), currencyPair);
+		  
+//	  } else if (args.length == 1 && args[0] instanceof Long){
+//		  GDAXTrades gdaxTrades = getGDAXTradesExtended(currencyPair, (Long)args[0], null);
+//		  
+//		  return GDAXAdapters.adaptTrades(gdaxTrades, currencyPair);		  
+		  
+	  } else if (args.length == 2 && args[0] instanceof Long && args[1] instanceof Long ){
+		  Long fromTradeId = (Long) args[0];
+		  Long toTradeId = (Long) args[1];
+		  
+		  log.debug("fromTradeId: {}, toTradeid: {}", fromTradeId, toTradeId);
+		  
+		  Long latestTradeId = toTradeId;
+		  
+		  GDAXTrades gdaxTrades = new GDAXTrades();
+		  
+		  for (;;) {	  
+			  GDAXTrades gdaxTradesNew = getGDAXTradesExtended(currencyPair, latestTradeId, 100); 
+			  gdaxTrades.addAll(gdaxTradesNew);		  
+			  
+			  log.debug("latestTradeId: {}, earliest-latest: {}-{}, trades: {}", 
+					  latestTradeId, gdaxTrades.getEarliestTradeId(), gdaxTrades.getLatestTradeId(), gdaxTrades);
+			  
+			  latestTradeId = gdaxTrades.getEarliestTradeId();
+			  
+			  if (gdaxTradesNew.getEarliestTradeId() == null) {break;}
+			  if (gdaxTrades.getEarliestTradeId() <= fromTradeId) {break;}
+		  }
+		  
+		  log.debug("earliest-latest: {}-{}", gdaxTrades.getEarliestTradeId(), gdaxTrades.getLatestTradeId());
+		  
+		  if (log.isDebugEnabled()) {
+		    gdaxTrades.stream().forEach(System.out::println);
```

Comment:
why do you use `System.out` and `e.printStackTrace' in the code if you have logging available?

Reply:
I'm torn here.... In this case just because it improves the readability of the log and is only for debugging to the console.  Though it prevents redirection of course.  I'm not fussed if you want me to change it.

I'll have to check out if there's a better way of handling the printstacktrace.

Reply:
I don't understand whats your point
SLF4J prints stracktraces when you do
`logger.debug("Some message about the context", exception);

Reply:
Fixed.